include/MySQL_HostGroups_Manager.h

```diff
@@ -1092,6 +1092,9 @@ class MySQL_HostGroups_Manager {
 	void set_server_current_latency_us(char *hostname, int port, unsigned int _current_latency_us);
 	unsigned long long Get_Memory_Stats();
 
+	int add_discovered_servers_to_mysql_servers_and_replication_hostgroups(vector<tuple<string, int, int>> new_servers);
```

Comment:
No reason to pass the vector by value, also, because it's never modified, could be easily converted into `const vector<tuple<string, int, int>>&`.